src/vm/eventpipe.h

```diff
@@ -114,6 +114,21 @@ class EventPipe
             InvokeCallback(eventPipeProviderCallbackData);
     }
 
+    // Returns the a number 0...N representing the processor number this thread is currently
+    // running on. If for any reason we can't tell then return 0xFFFFFFFF.
+    static unsigned int GetCurrentProcessorNumber()
+    {
+#ifndef FEATURE_PAL
+        if (s_pProcGroupOffsets)
+        {
+            PROCESSOR_NUMBER procNum;
+            GetCurrentProcessorNumberEx(&procNum);
```

Comment:
Is the performance of this method acceptable in all cases to call it unconditionally?

We have observed in the past that `GetCurrentProcessorNumber` is very slow in virtual machines because of it requires hypervisor trap.

Reply:
Ah interesting, I hadn't tested perf in a VM. Thanks for the heads up!

Given our limited remaining runway I'll find a way to mitigate this in 3.0 (probably off-by-default env var) and then we may be able to do better in the future.